### DISCUSSION TITLE
Play Framework 3.(0.5) Java and Pac4J [cross-posting]

### DISCUSSION THREAD

**Discussion by HenningHeinz** [playframework/playframework#12867] [Q&A] [answered]
Dear Community, 

I first posted this on 
https://stackoverflow.com/questions/79019063/the-latest-pac4j-play-framework-3-java-demo-does-not-compile-start

but the error message seem to be based on a change in Play Framework (3 Java). I am using Pac4J Security library and the recent demo application release does not run / compile on my JDK 17, Linux environment, Pekko engine).

```
[NoSuchMethodError: 'void play.api.mvc.EssentialFilter.$init$(play.api.mvc.EssentialFilter)']

org.pac4j.play.filters.SecurityFilter.<init>(SecurityFilter.scala:63)
org.pac4j.play.filters.SecurityFilter$$FastClassByGuice$$c834f8.GUICE$TRAMPOLINE(<generated>)
org.pac4j.play.filters.SecurityFilter$$FastClassByGuice$$c834f8.apply(<generated>)
     com.google.inject.internal.DefaultConstructionProxyFactory$FastClassProxy.newInstance(DefaultConstructionProxyFactory.java:82)
     com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:114)
     com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:91)
[...]
```

2.9.5 works with pac4j-play, but complains that it is missing the jakarta.x.validation class (documented in the migration guide though).

The problem should be reproducible by compiling the demo app from github.
https://github.com/pac4j/play-pac4j-java-demo

Any help/feedback is appreciated.

Thanks in advance.

Henning

**Comment by mkurz** [comment, marked as answer]
I am not sure why this is not working. First, the demo uses Scala 3.5 which Play does not support, so I downgraded to Scala 3.3.4 LTS but still the error occurs (also tried with 3.3.3 btw.). Maybe the library is compiled with a newer version of Scala therefore is incompatible with Play's artifacts? Not sure.

Meanwhile, because you try to use Play Java anyway, what you can do is just use Scala 2.13, this does actually work - see https://github.com/mkurz/play-pac4j-java-demo/commit/121ede6d94b241a21b1ec8d8d9e508c3d6c2b821
For you in the end it does not matter if you are using Scala 3 or 2 under the hood.

I will create an issue out of this thread so it gets tracked and a deeper look will be take what really is going on with Scala 3 here.

**Reply by mkurz** [reply]
- See #12870

**Reply by HenningHeinz** [reply]
Thank you very much Matthias. This helps me a lot.
